folium/map.py

```diff
@@ -480,3 +480,46 @@ def __init__(self, bounds, padding_top_left=None,
             padding_bottom_right=padding_bottom_right,
             padding=padding,
         )
+
+
+class CustomPane(MacroElement):
+    """
+    Creates a custom pane to hold map elements.
+
+    Behavior is as in https://leafletjs.com/examples/map-panes/
+
+    Parameters
+    ----------
+    name: string
+        Name of the custom pane. Other map elements can be added
+        to the pane by specifying the 'pane' kwarg when constructing
+        them.
+    z_index: int or string, default 625
+        The z-index that will be associated with the pane, and will
+        determine which map elements lie over/under it. The default
+        (625) corresponds to between markers and tooltips. Default
+        panes and z-indexes can be found at
+        https://github.com/Leaflet/Leaflet/blob/v1.0.0/dist/leaflet.css#L87
+    pointer_events: bool, default False
+        Whether or not layers in the pane should interact with the
+        cursor. Setting to False will prevent interfering with
+        pointer events associated with lower layers.
+    """
+    _template = Template(u"""
+        {% macro script(this, kwargs) %}
+            {{this._parent.get_name()}}.createPane('{{this.name}}');
+            {{this._parent.get_name()}}.getPane(
+                '{{this.name}}').style.zIndex = {{this.z_index}};
```

Comment:
You could use `tojson` here also instead of converting to str in the init... But this also works.